test/new-e2e/tests/discovery/docker_test.go

```diff
@@ -0,0 +1,93 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+package discovery
+
+import (
+	_ "embed"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/DataDog/datadog-agent/test/fakeintake/aggregator"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments"
+	awsdocker "github.com/DataDog/datadog-agent/test/new-e2e/pkg/provisioners/aws/docker"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/client/agentclient"
+	"github.com/DataDog/test-infra-definitions/components/datadog/dockeragentparams"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	pythonImage = "public.ecr.aws/docker/library/python:3"
+)
+
+type dockerDiscoveryTestSuite struct {
+	e2e.BaseSuite[environments.DockerHost]
+}
+
+func TestDiscoveryDocker(t *testing.T) {
+	agentOpts := []dockeragentparams.Option{
+		dockeragentparams.WithAgentServiceEnvVariable("DD_DISCOVERY_ENABLED", pulumi.StringPtr("true")),
+	}
+
+	e2e.Run(t,
+		&dockerDiscoveryTestSuite{},
+		e2e.WithProvisioner(
+			awsdocker.Provisioner(
+				awsdocker.WithAgentOptions(agentOpts...),
+			)))
+}
+
+func (s *dockerDiscoveryTestSuite) TestServiceDiscoveryContainerID() {
+	t := s.T()
+
+	client := s.Env().FakeIntake.Client()
+	err := client.FlushServerAndResetAggregators()
+	require.NoError(t, err)
+
+	s.assertDockerAgentDiscoveryRunning()
+
+	s.Env().RemoteHost.MustExecute("docker pull " + pythonImage)
```

Comment:
Can you make the test be skipped instead of failing if this command fails? We don't want spurious failures due to temporary network issues.